### DISCUSSION TITLE
Please help I am a total noob

### DISCUSSION THREAD

**Discussion by HighKitty** [PhlexPlexico/G5API#286] [Help]
I just can't make Get5 work.

I got a Server from linodes and a CS2 Server from dathost

No matter what I try if manually or with docker nothing seems to work for me.
Can someone please help me out and explain it step by step for someone that actually is not understanding anything.

**Comment by PhlexPlexico** [comment]
Get5 as a server plugin does not work with CS2. You would need a plugin such as MatchZy, or PugSharp.

In order to help, you would also need to provide log files, configs (with secrets removed) and information about your build environment (what OS, do you have a url/hostname for the machine, are you trying different ports)?

Thanks

**Comment by HighKitty** [comment]
Sry I ment the G5API and G5V to use it with MatchZy.

Like I said I have mostly no clue what I am doing and I got a Server from linodes for the webserver and installed Ubuntu on it, then I try to use SSH with the guides but nothing seem to work.

I tried to use Docker with the yml file but I don't even understand how to get it on the server and what I have to install on the server before hand etc.

I really want to use this but I feel kinda helples.

I don't even know what you mean with log files haha

I am not trying to insult or annoy people I just seek for help.

**Reply by PhlexPlexico** [reply]
Hi,

Okay, well you're going to need a domain(URL) in order to host your API (e.g. the old https://get5.phlexplexi.co) pointed to an IP address. I'm going to show a simple step-by-step for how to setup the Docker images, as it will be the easiest to work with.
1. Ensure that you have no webserver apps running on your current server (i.e. apache/nginx/caddy/etc).
2. Install [Docker](https://docs.docker.com/engine/install/ubuntu/). 
2a. Go get a domain and ensure it's pointing at your servers IP. If you do not wish to pay for a domain, use a service such as [DuckDNS](https://www.duckdns.org/). Follow their instructions on how to set this up.
3. Once that is installed, ensure it's working as intended (a quick `docker -v` should suffice).
4. Grab the config from [here](https://gist.github.com/PhlexPlexico/3d58b38f4b29cc540bb89be58cc702f7). *Do not* just paste and run, there are several values you need to adjust. Whereever you see anything that looks like `#CHANGEME` - change these to your specific values.

Once you save and close the compose file, simply call `docker compose up -d` to start the instance. It will pull all the images (Caddy for the web, the front-end, the API, and the database) and start the application for you. It will take a few minutes. If you need to reference the logs to see if it's running, simply type `docker container logs g5api`. If you wish to see the container names you can use `docker container ls -a`.

This project really is not a good starting point to try and setup, especially if you're brand new to this. Coming from experience of never really handling reverse proxies before this, it can be a headache to setup without Docker. Even with Docker it can be a bit troublesome. If you have any troubles, please *read* the logs and try and understand what is going wrong. If you have any troubles, feel free to post here, and post logs/configs (with secrets removed) and what you are trying to do. 

Thanks and good luck!

**Comment by HighKitty** [comment]
Oh boy thanks to you I am a lot further then I was before haha.

I have a new problem, I can't login through the "Sign in through Steam" button.
I hope I manage to attach picture now.
![2373397 1_image](https://github.com/PhlexPlexico/G5API/assets/74864982/d397147e-cd43-4477-93f3-0e8fae63f914)
![2360785 4_image](https://github.com/PhlexPlexico/G5API/assets/74864982/b9ed0a84-4526-4def-9f41-03f669144c1e)

**Comment by HighKitty** [comment]
I also found this in the G5API log

![image](https://github.com/PhlexPlexico/G5API/assets/74864982/8faf5c33-4ffb-4cda-9784-ad009e3b37f6)

**Comment by HighKitty** [comment]
Never mind I made it work with the  - LOCALLOGINS= command

I put it in true and I can atleast use it now. Works like a sharm, thank you for creating this its amazing!

**Reply by PhlexPlexico** [reply]
Just as an aside, steam OAuth is failing because your system clock isn't timed correctly with Steam's clock. You need to make sure your servers system clock is in sync (usually via NTP).

Thanks!

**Reply by HighKitty** [reply]
Can you maybe assist me in how to do it? Because as usually I have no clue what you talking about.

**Reply by PhlexPlexico** [reply]
`sudo timedatectl set-ntp on` 

If that doesn't work, refer to this here https://www.digitalocean.com/community/tutorials/how-to-set-up-time-synchronization-on-ubuntu-20-04

**Reply by PhlexPlexico** [reply]
Alternatively, use this point of the article https://www.digitalocean.com/community/tutorials/how-to-set-up-time-synchronization-on-ubuntu-20-04#switching-to-ntpd

**Reply by HighKitty** [reply]
So it says System clock synchronized: ye

ntpsec.service is Active and running

but NTP Server n/a and  not supported? wut

![image](https://github.com/PhlexPlexico/G5API/assets/74864982/b196f909-e007-463b-b554-1a0a57410b00)

**Reply by PhlexPlexico** [reply]
Okay, so you can't be running `ntpd` and `timedatectl` at the same time. Try removing `ntpd` by calling `sudo apt remove ntpd` then after that call `sudo timedatectl set-ntp on`.

**Reply by HighKitty** [reply]
I get "E: Unable to locate package ntpd"

**Reply by PhlexPlexico** [reply]
Okay, well whichever packaged you ended up installing, you will need to uninstall that, and then call`sudo timedatectl set-ntp on` afterwards. Would that be `ntpsec`?

**Reply by HighKitty** [reply]
The only thing I did install was ntp with sudo apt-get install ntp

**Reply by HighKitty** [reply]
Okay I think I got it working.

![image](https://github.com/PhlexPlexico/G5API/assets/74864982/c62c618a-4a03-4420-9424-48e3f465d65c)

Can't login through Steam still tho.

**Reply by PhlexPlexico** [reply]
Okay so odd question - did you provide a steam API key in the docker image? And make sure you restart the application as well. `docker compose stop; docker compose up -d`.

**Reply by HighKitty** [reply]
![9161598 6_image](https://github.com/PhlexPlexico/G5API/assets/74864982/dc8e7857-fc51-42d4-960f-9fca18f8af23)

Yes

**Reply by PhlexPlexico** [reply]
Most of these issues when it comes to failing to verify assertion is on the server side. You'll need to try a few different things like flushing the DNS cache and rebooting the server. you could also just take your docker compose file and try starting it on a different server (that has NTP synced, etc).

**Reply by PhlexPlexico** [reply]
Could you also use gist.github.com and paste your current docker-compose with your secrets removed? (db password, redis password, steam API key, etc). I would like to see the URLs you're using as well, to ensure the callback is setting the correct URL.

**Reply by HighKitty** [reply]
https://gist.github.com/HighKitty/f513eca01f3144e13117587934d58598

**Reply by PhlexPlexico** [reply]
Hmm okay, so the config is looking correct. Now it's probably just a server issue, so whether that be timing, or if Steam has blocked the IP due to too many failures, or if the server provider does not offer the right firewall rules. I'm not sure if there's much else I can do to help, sorry!

**Reply by HighKitty** [reply]
Does some ports have to be open for the API?

**Reply by PhlexPlexico** [reply]
It should only be 443 and 80, since the API is reverse proxied.

**Reply by HighKitty** [reply]
Hmm ye Firewall should not be an issue ports are open.
Maybe Server just to slow? specs are not the best tbh.

But you have been a massive help nontheless I mean I can still use the entire thing with the local login so I am super grateful <3

**Reply by HighKitty** [reply]
Wait I actually have a question.

The API Key I use the same one from the CS2 Server I just noticed its called GSLT and now I am afraid I actually DON'T use a Steam API Key.

Is there a chance to change the API without reinstalling the entire thing?

**Reply by PhlexPlexico** [reply]
https://steamcommunity.com/dev/apikey

This is where you get the steam API key. Once you fill in the proper API key in your docker-compose file use `docker compose stop; docker compose up -d` to change the variable. This can be checked after by doing a `docker container exec -it g5api sh` (it's either that or `sh g5api`) and then navigate to `/Get5Api/config/production.json` to view your config in the docker image.

**Reply by HighKitty** [reply]
Man you are THE man how can I thank you :')

**Reply by PhlexPlexico** [reply]
No need, just glad you got it up and running! Cheers! :)
